### PR TITLE
Fix of carrier options in Shipping > Preferences + Minor fix on multiple options of carrier choice

### LINKED ISSUE #20023: Same name carriers don't show in Shipping > Preferences > Default carrier
#### Describe the bug

There are missing carriers in the list of "**Shipping > Preferences > Carrier options > Default carrier**", the problem occurs with carriers with the same name.

#### Expected behavior

Display the complete list of carriers availables.

#### Steps to Reproduce

1. Create more than one carrier with the same name
2. In the carrier selection in "**Shipping > Preferences > Carrier options > Default carrier**" will display only one

**Screenshots**

error-01 
error-02

#### Additional information

* PrestaShop version: Develop / 1.7.7.x
* PHP version: 7.2

### PR DESCRIPTION
**Pull request by ghost**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop / 1.7.7.x
| Description?  | This is a complement fix of our previous PR https://github.com/PrestaShop/PrestaShop/pull/12379.
| Type?         | bug fix
| Category?     | BO
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | #20023
| How to test?  | Create more than one carrier with the same name and in the carrier selection at **Shipping > Preferences > Carrier options > Default carrier** will display only one. 
error-01 
error-02

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/19971)

### CODE REVIEW

**Comment by Progi1984** [comment]
@Rolige  Thank you for your contribution. :hugs: 

Would you mind to rebase your pull request to remove Merge Commit ? 
Could you explain in [an issue your initial bug for this PR](https://github.com/PrestaShop/PrestaShop/issues/new/choose) (or may be link to an issue) ?

Thanks :+1:

**Comment by Rolige** [comment]
> @Rolige Thank you for your contribution. 🤗
> 
> Would you mind to rebase your pull request to remove Merge Commit ?
> Could you explain in [an issue your initial bug for this PR](https://github.com/PrestaShop/PrestaShop/issues/new/choose) (or may be link to an issue) ?
> 
> Thanks 👍

Issue created! About to rebase, do you have a tutorial of how to achieve this with a forked repo maintained with "upstream"?

**Comment by Matt75** [comment]
Hi @Rolige here some commands to syncing your fork : https://devdocs.prestashop.com/1.7/contribute/contribute-pull-requests/create-pull-request/#syncing-your-fork

File: controllers/admin/AdminCartRulesController.php [hunk 1]
```diff
@@ -671,7 +671,7 @@ public function renderForm()
                             $value = $carrier['id_carrier'] . ' - ' . Configuration::get('PS_SHOP_NAME');
                         } else {
                             $value = $carrier['id_carrier'] . ' - ' . $carrier['name'];
-                            if ($carrier['name']) {
+                            if ($carrier['delay']) {
```
Comment (PierreRambaud) [inline controllers/admin/AdminCartRulesController.php:674]: Just think about the delay, is it possible to have `0` as value?
Shouldn't be something like:
```suggestion
                            if (isset($carrier['delay'])) {
```

File: src/Adapter/Carrier/CarrierDataProvider.php [hunk 2]
```diff
@@ -83,7 +83,12 @@ public function getActiveCarriersChoices($languageId = null)
         $carriersChoices = [];
 
         foreach ($carriers as $carrier) {
-            $carriersChoices[$carrier['name']] = $carrier['id_carrier'];
+            $choiceId = $carrier['id_carrier'] . ' - ' . $carrier['name'];
+            if ($carrier['delay']) {
```
Comment (PierreRambaud) [inline src/Adapter/Carrier/CarrierDataProvider.php:87]:
```suggestion
            if (isset($carrier['delay'])) {
```

File: src/Core/Form/ChoiceProvider/CarrierByReferenceChoiceProvider.php [hunk 3]
```diff
@@ -73,7 +73,7 @@ public function getChoices()
 
         foreach ($carriers as $carrier) {
             $choiceId = $carrier['id_carrier'] . ' - ' . $carrier['name'];
-            if ($carrier['name']) {
+            if ($carrier['delay']) {
```
Comment (PierreRambaud) [inline src/Core/Form/ChoiceProvider/CarrierByReferenceChoiceProvider.php:76]:
```suggestion
            if (isset($carrier['delay'])) {
```

File: src/PrestaShopBundle/Form/Admin/Product/ProductShipping.php [hunk 4]
```diff
@@ -69,7 +69,7 @@ public function __construct($translator, $legacyContext, $warehouseDataProvider,
         $this->carriersChoices = [];
         foreach ($carriers as $carrier) {
             $choiceId = $carrier['id_carrier'] . ' - ' . $carrier['name'];
-            if ($carrier['name']) {
+            if ($carrier['delay']) {
```
Comment (PierreRambaud) [inline src/PrestaShopBundle/Form/Admin/Product/ProductShipping.php:72]:
```suggestion
            if (isset($carrier['delay'])) {
```

**Review by PierreRambaud — COMMENTED** [review]
Hi @Rolige 
First of all, we are sorry for the delay.
I review your code with only one question, if you have time to answer and rebase your PR :pray:

**Comment by PierreRambaud** [comment]
Hi @Rolige,

Since we had no news from you for more than 30 days, I'll close this pull request. Feel free to reopen or open another one if you think it's still relevant.

Thanks!

**Pull request closed without merging**